client/ayon_maya/api/lib.py

```diff
@@ -4343,3 +4343,39 @@ def get_sequence(filepath, pattern="%04d"):
         os.path.join(source_dir, filename)
         for filename in collections[0]
     ]
+
+
+@contextlib.contextmanager
+def write_face_sets_for_materials(shapes):
+    """Write material per face sets in the meshes during context
+    Activated only when `writeFaceSets` alembic options enabled.
+
+    Args:
+        shapes (List[str]): The shapes to add into face sets for any component
+            assignments of shading engines.
+    """
+    original_material_list = {}
+    for shape in shapes:
+        shading_engines = cmds.listConnections(shape, destination=True, type="shadingEngine") or []
+        original_material_list.update({shape: shading_engines})
+
+        for shading_engine in shading_engines:
+            cmds.hyperShade(objects=shading_engine)
+            #store a list of the selected objects
+            object_with_mat = cmds.ls(selection=True)
+            #assign lambert1 (a material that should always exist) to selection
+            cmds.hyperShade(assign="lambert1")
+            #clear selection
+            cmds.select(deselect=True)
+
+            for shape in object_with_mat:
+                cmds.select(shape +".f[*]",add=True)
+            cmds.hyperShade(assign=shading_engine)
```

Comment:
Sorry - wait a second. What are we doing here, and why? What!?

1. Lambert1 is not required to exist, especially now in recent Maya versions where it defaults to a standard surface.
2. Why does this use a selection? What?
3. Do not use the `hypershade` command for what is essentially an assignment of a set.

What was wrong with the previous implementation where it generates object sets?